var/spack/repos/builtin/packages/mpich/package.py

```diff
@@ -96,8 +116,14 @@ class Mpich(AutotoolsPackage):
 
     depends_on('ucx', when='netmod=ucx')
 
-    depends_on('libpciaccess', when="+pci")
-    depends_on('libxml2')
+    # The dependencies on libpciaccess and libxml2 come from the embedded
+    # hwloc, which, before version 3.3, was used only for Hydra.
+    depends_on('libpciaccess', when="@:3.2+hydra+pci")
+    depends_on('libxml2', when='@:3.2+hydra+libxml2')
```

Comment:
Does `config_args += self.enable_or_disable('libxml2')` with `+libxml2` (i.e. `--enable-libxml2`) prevent the embedded `hwloc` from building its own `libxml2`?

`depends_on('libxml2'...` will cause Spack to build a separate instance of `libxml2`, so if If the embedded `hwloc` has its own `libxml2`, then there may be a conflict.

In other words: If the embedded `hwloc` always builds `libxml2` then I think there would be a conflict in any case that we add `libxml2` as a Spack dependency.